src/app/common/dialogs/export-pdf-dialog/export-pdf-dialog.component.ts

```diff
@@ -61,12 +67,17 @@ export class ExportPdfDialog extends BaseDialog {
 		this.badge = badge;
 		this.showModal();
 
-		this.generateSingleBadgePdf(this.badge, markdown);
+		this.pdfService.getPdf(badge.slug).subscribe((url) => {
+			this.pdfSrc = url;
+		})
 
-		return new Promise<void>((resolve, reject) => {
-			this.resolveFunc = resolve;
-			this.rejectFunc = reject;
-		});
+		
+		// this.generateSingleBadgePdf(this.badge, markdown);
```

Comment:
Add a comment here to show this is the old implementation or remove it if no longer used.